apps/coscrad-frontend/src/components/resources/terms/term-index.presenter.tsx

```diff
@@ -70,33 +70,94 @@ export const TermIndexPresenter = (termsIndexResult: TermIndexState) => {
-        name: doesSomeMultilingualTextItemInclude,
-        vocabularyLists: (vocabularyLists: IVocabularyListRecordForTerm[], searchTerm: string) =>
-            vocabularyLists.some(({ name }) =>
-                name.items.some(({ text }) => doesTextIncludeCaseInsensitive(text, searchTerm))
-            ),
-        tokens: (tokens, searchTerm) =>
-            (tokens || []).some(({ characters }) =>
-                characters.some((c) => {
-                    const doesMatch = c.text === searchTerm.toLowerCase();
+    const onSearch = (scope: IndexSearchScope<ITermViewModel>, queryFromForm: string) => {
+        if (!isNonEmptyString(queryFromForm)) {
+            return; // TODO fetch with no filters
+        }
 
-                    if (c.isOutOfAlphabet) return false;
+        let condition;
 
-                    return doesMatch;
-                })
-            ),
+        if (queryFromForm.charAt(0) === '{' && queryFromForm.includes('}')) {
+            const extractedLanguageCode = queryFromForm.slice(1).split('}')[0];
+
+            if (Object.values(LanguageCode).some((lc) => lc === extractedLanguageCode)) {
+                condition = {
+                    type: 'SIMPLE',
+                    operator: 'MULTILINGUAL_TEXT_INCLUDES',
+                    params: [queryFromForm, extractedLanguageCode],
+                };
+            }
+        }
+
+        // default
+        if (!condition) {
+            condition = {
+                type: 'SIMPLE',
+                operator: 'MULTILINGUAL_TEXT_INCLUDES',
+                params: [queryFromForm],
+            };
+        }
+
+        // const filter = {
+        //     type: 'OR',
+        //     conditions: (scope === ALL_PROPERTIES_SEARCH_KEY
+        //         ? ['name', 'contributions', 'vocabularyLists', 'tokens']
+        //         : [scope]
+        //     ).map((field) => ({
+        //         ...condition,
+        //         field,
+        //     })),
+        // };
+
+        // TODO type safety
+        const action = filterTerms({ scope, query: queryFromForm });
+
+        console.log({
+            dispatching: action,
+        });
+
+        dispatch(
+            // fetchTerms({
+            //     filter: filter,
+            //     pagination: {
+            //         size: 100,
+            //         page: 1,
+            //     },
+            // })
+            action
+        );
     };
 
     return (
-        <IndexTable
-            type={AggregateType.term}
-            headingLabels={headingLabels}
-            tableData={terms as ITermViewModel[]}
-            cellRenderersDefinition={cellRenderersDefinition}
-            heading={'Terms'}
-            filterableProperties={['name', 'contributions', 'vocabularyLists', 'tokens']}
-            matchers={matchers}
-        />
+        <Stack>
+            {/* TODO Pull this from the resource info \ config? */}
+            <Typography variant="h2">{'Terms'}</Typography>
```

Comment:
I noticed when testing with Cypress that the label `Terms` was showing up twice on the screen. Is this still the case?